patches/@tanstack+react-query+4.2.1.patch

```diff
@@ -0,0 +1,14 @@
+diff --git a/node_modules/@tanstack/react-query/build/esm/index.js b/node_modules/@tanstack/react-query/build/esm/index.js
+index 80f4739..6fdadf8 100644
+--- a/node_modules/@tanstack/react-query/build/esm/index.js
++++ b/node_modules/@tanstack/react-query/build/esm/index.js
+@@ -2496,7 +2496,8 @@ class QueryObserver extends Subscribable {
+     const result = {
+       status,
+       fetchStatus,
+-      isLoading: status === 'loading',
++      isLoading: status === 'loading' && fetchStatus === 'fetching',
++      isIdle: status === 'loading' && fetchStatus === 'idle',
```

Comment:
Might be a couple silly questions here, I might be confused 😛 

The change to `isLoading` makes sense to me, since `status === loading` only on initial load and a cold cache, right?

But "idle" to me means "not fetching atm". With `status === loading` here, won't this only be idle on initial load as well? Should it be `status === 'success' && fetchStatus === 'idle'`?

Seems like a weird choice by our boy Tanner, but like I said, might just not grok this bit yet 😅

Reply:
> But "idle" to me means "not fetching atm". With status === loading here, won't this only be idle on initial load as well? Should it be status === 'success' && fetchStatus === 'idle'?

The terminology of `status` is a bit confusing, but it essentially means "cache status". A `status` of `loading` means we do not have any data in the cache. We will consider the state being idle when we do not have any data in the cache, and we are not fetching.

Reply:
Ok my mistake, I misunderstood 👍 dig the patch too!